lib/dpul_collections/utilities.ex

```diff
@@ -0,0 +1,36 @@
+defmodule DpulCollections.Utilities do
+  alias DpulCollections.IndexingPipeline
+  alias DpulCollections.IndexingPipeline.Figgy
+
+  def reindex_solr(cache_version) do
+    indexing_processor_marker =
+      IndexingPipeline.get_processor_marker!("figgy_indexer", cache_version)
+
+    IndexingPipeline.delete_processor_marker(indexing_processor_marker)
+    # When the consumer is stopped, it supervisor restarts it immediately.
+    # This resets the state and last queried marker.
+    GenServer.stop(Figgy.IndexingConsumer)
+  end
+
+  def reindex_transformation_cache(cache_version) do
+    transformation_processor_marker =
+      IndexingPipeline.get_processor_marker!("figgy_transformer", cache_version)
+
+    IndexingPipeline.delete_processor_marker(transformation_processor_marker)
+    GenServer.stop(Figgy.TransformationConsumer)
+  end
+
+  def reindex_hydration_cache(cache_version) do
+    hydration_processor_marker =
+      IndexingPipeline.get_processor_marker!("hydrator", cache_version)
+
+    IndexingPipeline.delete_processor_marker(hydration_processor_marker)
+    GenServer.stop(Figgy.HydrationConsumer)
+  end
+
+  def reindex_all(cache_version) do
```

Comment:
Do we need to reset the state of everything if want to reindex? Like if we want a full re-hydration, can't we just reset the hydrator? And if we want a full retransformation without hydration, we just reset the transformer?